changes.go

```diff
@@ -22,54 +22,294 @@ func FromData(data *charm.BundleData) []*Change {
 }
 
 // Change holds a single change required to deploy a bundle.
-type Change struct {
-	// Id is the unique identifier for this change.
-	Id string `json:"id"`
-	// Method is the action to be performed to apply this change.
-	Method string `json:"method"`
-	// Args holds a list of arguments to pass to the method.
-	Args []interface{} `json:"args"`
-	// Requires holds a list of dependencies for this change. Each dependency
+type Change interface {
+	// Id returns the unique identifier for this change.
+	Id() string
+	// Requires returns a list of dependencies for this change. Each dependency
```

Comment:
// Requires holds the ids of all the changes that must
// be applied before this one.

?